### PR TITLE
[example] llama2 add fine-tune example

### PR DESCRIPTION
**Pull request by flybird11111**
## 📌 Checklist before creating the PR

- [x] I have created an issue for this PR for traceability
- [x] The title follows the standard format: `[doc/gemini/tensor/...]: A concise description`
- [x] I have added relevant tags if possible for us to better distinguish different PRs

## 🚨 Issue number

> Link this PR to your issue with words like fixed to automatically close the linked issue upon merge
>
> e.g. `fixed #1234`, `closed #1234`, `resolved #1234`

#4643 
#4705 

## 📝 What does this PR do?

> Summarize your work here.
> if you have any plots/diagrams/screenshots/tables, please attach them here.

llama2 add fine-tune example

## 💥 Checklist before requesting a review

- [x] I have linked my PR to an issue ([instruction](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue))
- [x] My issue clearly describes the problem/feature/proposal, with diagrams/charts/table/code if possible
- [x] I have performed a self-review of my code
- [x] I have added thorough tests.
- [ ] I have added docstrings for all the functions/methods I implemented

## ⭐️ Do you enjoy contributing to Colossal-AI?

- [ ] 🌝 Yes, I do.
- [ ] 🌚 No, I don't.

Tell us more if you don't enjoy contributing to Colossal-AI.

### CODE REVIEW

**Comment by adivoj** [comment]
Hi and thanks for the efforts, 

I'd really like to use it to finetune 7b on my T4 but I get OOM.

Is it possible to quantize it?

BitsAndBytes, for example: https://github.com/tloen/alpaca-lora/issues/486
or
AutoGPTQ, for example: https://huggingface.co/blog/gptq-integration#fine-tune-quantized-models-with-peft

All the best!

**Comment by flybird11111** [comment]
> quantize

Thank you, we will consider your suggestions.

**Review by FrankLeeeee — APPROVED** [review]

File: examples/language/llama2/requirements.txt [hunk 1]
```diff
@@ -1,4 +1,4 @@
-colossalai>=0.3.0
+colossalai==0.3.2
```
Comment (Fridge003) [inline examples/language/llama2/requirements.txt:1]: I don't know whether fixing the version of colossalai is appropriate. Some bugs will be fixed in the next released version.

File: examples/language/llama2/scripts/finetune_7B/finetune.sh [hunk 2]
```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+
+################
+#Load your environments and modules here
+################
+
+HOSTFILE=$(realpath hosts.txt)
+
+cd ../..
```
Comment (Fridge003) [inline examples/language/llama2/scripts/finetune_7B/finetune.sh:9]: Please describe the usage of finetune.sh in README, user might stay in the root directory of llama example.

File: examples/language/llama2/scripts/pretrain_7B/pretrain.sh [hunk 3]
```diff
@@ -0,0 +1,13 @@
+#!/bin/bash
+
+################
+#Load your environments and modules here
+################
+
+HOSTFILE=$(realpath hosts.txt)
+
+cd ../..
```
Comment (Fridge003) [inline examples/language/llama2/scripts/pretrain_7B/pretrain.sh:9]: Same problem

File: examples/language/opt/requirements.txt [hunk 4]
```diff
@@ -1,4 +1,4 @@
-colossalai >= 0.1.12
+colossalai == 0.3.2
```
Comment (Fridge003) [inline examples/language/opt/requirements.txt:1]: Same problem.

**Review by Fridge003 — APPROVED** [review]

**Pull request merged**